src/peer_sourcehash.cc

```diff
@@ -125,7 +125,7 @@ peerSourceHashInit(void)
 
     for (k = 1; k <= K; ++k) {
         double Kk1 = (double) (K - k + 1);
-        p = sourcehash_peers[k - 1];
+        auto p = sourcehash_peers[k - 1];
```

Comment:
```suggestion
        const auto p = sourcehash_peers[k - 1];
```

Please check for other candidates as well.